pkg/types/relayer.go

```diff
@@ -2,11 +2,71 @@ package types
 
 import (
 	"context"
+	"fmt"
 	"math/big"
+	"regexp"
 
 	"github.com/google/uuid"
 )
 
+const (
+	NetworkEVM      = "evm"
+	NetworkCosmos   = "cosmos"
+	NetworkSolana   = "solana"
+	NetworkStarkNet = "starknet"
+)
+
+var SupportedRelays = map[string]struct{}{
+	NetworkEVM:      {},
+	NetworkCosmos:   {},
+	NetworkSolana:   {},
+	NetworkStarkNet: {},
+}
+
+type RelayID struct {
+	Network string
+	ChainID string
+}
+
+// ID uniquely identifies a relayer by network and chain id
+func (i *RelayID) Name() string {
+	return fmt.Sprintf("%s.%s", i.Network, i.ChainID)
+}
+
+func (i *RelayID) String() string {
+	return i.Name()
+}
+func NewRelayID(n string, c string) RelayID {
+	return RelayID{Network: n, ChainID: c}
+}
+
+var idRegex = regexp.MustCompile(
+	fmt.Sprintf("^((%s)|(%s)|(%s)|(%s))\\.", NetworkEVM, NetworkCosmos, NetworkSolana, NetworkStarkNet),
```

Comment:
as said below, i think the regex is more complex than needed. however, if you need to keep to it, then it should be constructed from SupportNetwork rather than individual consts

Reply:
@krehermann Do you mind if I defer this? This is just the existing implementation copied from core to common